### ISSUE TITLE
Linux xboard Mayhemduck problems

### ISSUE THREAD

**Issue by RickGroszkiewicz** [SamuraiDangyo/mayhem#7]
I have been trying to test MayhemDuck on an old linux laptop. I can't get Mayhemduck to load at all:
[https://talkchess.com/forum3/viewtopic.php?f=7&t=80829&sid=274877d988b510981186cf465351d6f6&start=180](url)

Maybe it is because my laptop is ancient. I don't see the source for Mayhemduck. Any chance you could compile Mayhemduck using "make oldcpu"

**Comment by SamuraiDangyo** [comment]
Hello.
Thanks for the report. I looked at the issue. Xboard and UCI2WB compile well but simply send illegal moves.
I made MayhemDuck to write all commands to a file and this is the result
See:
```
[Event "Computer Chess Game"]
[Site "th-desktop"]
[Date "2023.03.05"]
[Round "-"]
[White "MayhemDuck 2.0 (UCI2WB)"]
[Black "human"]
[Result "*"]
[TimeControl "180+1"]
[Variant "duck"]
[Annotator "1. +0,34"]

1. d4,c6 {+0,34/6} Nf6,f3 2. Nc3,c6 {+0,40/6 8} d5,f3 3. Bg5,d7 {+0,40/5 7}
Nc6,f3

-------------------------------

### MayhemDuck Commands ###

CMD: uci
CMD: isready
CMD: ucinewgame
CMD: isready
CMD: setoption name UCI_Variant value duck
CMD: ucinewgame
CMD: isready
CMD: position startpos moves
CMD: go btime 180000 wtime 180000 winc 1000 binc 1000
CMD: position startpos moves d2d4,d4c6 g8f6,f3g6
CMD: go btime 174660 wtime 173080 winc 1000 binc 1000
CMD: position startpos moves d2d4,d4c6 g8f6,f3g6 b1c3,c3c6 d7d5,f3c5
CMD: go btime 168620 wtime 166420 winc 1000 binc 1000
CMD: position startpos moves d2d4,d4c6 g8f6,f3g6 b1c3,c3c6 d7d5,f3c5 c1g5,g5d7 b8c6,f3b6
CMD: go btime 160480 wtime 160020 winc 1000 binc 1000
CMD: quit
```

**Comment by RickGroszkiewicz** [comment]
Yes, that sounds correct.  HGM is currently studying the UCI2WB code, since that seems to be the culprit. 

But that problem does not occur when I play against mayhemduck. Any chance you can compile a linux version so I could play against mayhemduck on my very old laptop.

**Comment by RickGroszkiewicz** [comment]
HGM found that we were not using his latest updates for xboard. After testing today, mayhemduck versions before 2.0 work properly.

Version 2.0 of mayhemduck still crashes when loaded to xboard.

**Comment by SamuraiDangyo** [comment]
Hello. Yes 2.0 crashes duno why tho. Fixed in 2.1 which I'll upload shortly...
